e2e/vmss.go

```diff
@@ -90,40 +105,167 @@ func skipTestIfSKUNotAvailableErr(t *testing.T, err error) {
 	}
 }
 
-func cleanupVMSS(ctx context.Context, t *testing.T, vmssName string, cluster *Cluster, privateKeyBytes []byte) {
+func cleanupVMSS(ctx context.Context, s *Scenario) {
 	// original context can be cancelled, but we still want to collect the logs
-	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), 3*time.Minute)
+	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), 5*time.Minute)
+	defer cancel()
+	defer deleteVMSS(ctx, s)
+
+	if s.VHD.Windows() {
+		extractLogsFromWindowsVM(ctx, s)
+		return
+	}
+
+	logFiles, err := extractLogsFromVM(ctx, s)
```

Comment:
same, could we embed the logic for linux vs windows inside the `extractLogsFromVM`

I would prefer to have `extractLogsFromWindowsVM` and `extractLogsFromLinuxVM` OR even better just `extractLogsFromVM` which uses the Scenario `s` to decide internally how to extract the logs. 
```
extractLogsFromVM() {

if s.VHD.windows
  extractLogsFromWindowsVM
else
  extractLogsFromLinuxVM

}